app/gateway/traffic-control/proxying.md

```diff
@@ -27,25 +27,297 @@ works_on:
+
+1. The number of retries. This can be configured per Service using the [`retries`](/gateway/entities/service/#schema-service-retries) property.
+1. What exactly constitutes an error. Here {{site.base_gateway}} uses the Nginx defaults, which means an error or timeout that occurs while establishing a connection with the server, passing a request to it, or reading the response headers. 
+This is based on Nginx's [`proxy_next_upstream`](https://nginx.org/docs/http/ngx_http_proxy_module.html#proxy_next_upstream) directive. 
+This option is not directly configurable through {{site.base_gateway}}, but can be added using a custom Nginx configuration. 
+See the [Nginx directives reference][/gateway/nginx-directives/] for more details.
+
+## Response
+
+{{site.base_gateway}} receives the response from the upstream service and sends it back to the downstream client in a streaming fashion. 
+At this point, {{site.base_gateway}} executes subsequent plugins added to the Route or Service that implement a hook in the `header_filter` phase.
+
+Once the `header_filter` phase of all registered plugins has been executed, the following headers are added by {{site.base_gateway}} and the full set of headers is sent to the client:
+
+{% table %}
+columns:
+  - title: Header
+    key: header
+  - title: Description
+    key: description
+rows:
+  - header: "`Via: kong/x.x.x`"
+    description: "`x.x.x` is the {{site.base_gateway}} version in use."
+  - header: "`X-Kong-Proxy-Latency: <latency>`"
+    description: "`latency` is the time, in milliseconds, between {{site.base_gateway}} receiving the request from the client and sending the request to your upstream service."
+  - header: "`X-Kong-Upstream-Latency: <latency>`"
+    description: "`latency` is the time, in milliseconds, that {{site.base_gateway}} was waiting for the first byte of the upstream service response."
+{% endtable %}
+
+Once the headers are sent to the client, {{site.base_gateway}} starts executing plugins for the Route or Service that implement the
+`body_filter` hook. 
+This hook may be called multiple times, due to the streaming nature of Nginx. 
+Each chunk of the upstream response that is successfully processed by such `body_filter` hooks is sent back to the client.
+
+## Proxy WebSocket traffic
+
+{{site.base_gateway}} supports WebSocket traffic thanks to the underlying Nginx implementation.
+When you want to establish a WebSocket connection between a client and your upstream services through {{site.base_gateway}}, you must establish a WebSocket handshake.
+This is done via the HTTP Upgrade mechanism. 
+This is what your client request made to {{site.base_gateway}} would look like:
+```
+GET / HTTP/1.1
+Connection: Upgrade
+Host: my-websocket-api.com
+Upgrade: WebSocket
+```
+
+This makes {{site.base_gateway}} forward the `Connection` and `Upgrade` headers to your upstream service, instead of dismissing them due to the hop-by-hop nature of a standard HTTP proxy.
```

Comment:
```suggestion
This configures {{site.base_gateway}} to forward the `Connection` and `Upgrade` headers to your upstream service instead of dismissing them due to the hop-by-hop nature of a standard HTTP proxy.
```